src/axom/inlet/tests/inlet_function.cpp

```diff
@@ -139,6 +139,84 @@ TEST(inlet_function, simple_double_to_double_through_table)
   EXPECT_FLOAT_EQ(result, (arg * 3.4) + 9.64);
 }
 
+TEST(inlet_function, simple_void_to_double_through_table)
+{
+  std::string testString = "function foo () return 9.64 end";
+  DataStore ds;
+  auto inlet = createBasicInlet(&ds, testString);
+
+  inlet.addFunction("foo", FunctionTag::Double, {}, "foo's description");
+
+  auto callable = inlet["foo"].get<std::function<FunctionType::Double()>>();
+  double result = callable();
+  EXPECT_FLOAT_EQ(result, 9.64);
+}
+
+TEST(inlet_function, simple_double_to_void_through_table)
+{
+  // Test a function that returns nothing by using it to modify a global
+  std::string testString = "bar = 19.9; function foo (a) bar = a end";
+  DataStore ds;
+  auto inlet = createBasicInlet(&ds, testString);
+
+  inlet.addFunction("foo",
+                    FunctionTag::Void,
+                    {FunctionTag::Double},
+                    "foo's description");
+
+  auto callable =
+    inlet["foo"].get<std::function<FunctionType::Void(FunctionType::Double)>>();
+  double arg = -6.37;
+  callable(arg);
+
+  inlet.addDouble("bar", "bar's description");
+  double result = inlet["bar"];
+  EXPECT_FLOAT_EQ(result, arg);
```

Comment:
Suggestion: It might be good to check that `inlet["bar"] == 19.9` before `callable(arg)`

Reply:
I agree that this would be useful to check, but currently Inlet will only read something from the input file once (that is, later updates will not affect the Inlet state)  - this is why I had to delay the `addDouble` call until after the function is invoked.